pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -215,9 +222,39 @@ func (m *kubeGenericRuntimeManager) calculateLinuxResources(cpuRequest, cpuLimit
 		resources.CpuPeriod = cpuPeriod
 	}
 
+	if supportsMemoryOOMGroup() {
+		// Turn on memory.oom.group which tells the kernel to treat the entire container cgroup as
+		// a single unit for OOM kill purposes.  This will cause all processes in the container to be
+		// OOM killed together if any are OOM killed.
+		resources.Unified = map[string]string{
+			"memory.oom.group": "1",
```

Comment:
There should probably be some setting on the container to let users opt out of this behavior, but I didn't see any good place to hang it.  Suggestions?

Reply:
I think making it default is fine and non-controversial